.github/workflows/upgrade.yml

```diff
@@ -213,9 +213,19 @@ jobs:
             DEPLOY_FLAGS='--trust'
           fi
           
-          juju deploy ${CHARM_${{ matrix.cloud }}} $DEPLOY_FLAGS
+          CURRENT_CHARM=${CHARM_${{ matrix.cloud }}}
           
-          juju wait-for application ${CHARM_${{ matrix.cloud }}}
+          juju deploy ${CURRENT_CHARM} $DEPLOY_FLAGS
```

Comment:
The application name is not guaranteed to be the same as the charm name, to be super safe you could specify the application name. E.g.
```
juju deploy ${CURRENT_CHARM} ${CURRENT_CHARM} $DEPLOY_FLAGS
```

Reply:
I like being super safe, thanks !